blockauth/apple/notification_service.py

```
@@ -55,6 +62,74 @@ def _parse_events_claim(raw: Any) -> dict[str, Any]:
+            "sub": events.get("sub"),
+        }
+    digest = hashlib.sha256(
+        json.dumps(replay_identity, default=str, separators=(",", ":"), sort_keys=True).encode()
+    ).hexdigest()
+    return f"{_REPLAY_CACHE_KEY_PREFIX}{digest}"
+
+
+def _reject_replay(claims: dict[str, Any], events: dict[str, Any]) -> str:
+    ttl_seconds = _max_age_seconds() + _future_leeway_seconds()
+    cache_key = _replay_cache_key(claims, events)
+    if not default_cache.add(cache_key, True, timeout=ttl_seconds):
```

Ruff (0.15.12)

[warning] Boolean positional value in function call

(FBT003)